src/compiler/checker.ts

```diff
@@ -17395,7 +17389,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return (type as SubstitutionType).objectFlags & ObjectFlags.IsGenericType;
         }
         return (type.flags & TypeFlags.InstantiableNonPrimitive || isGenericMappedType(type) || isGenericTupleType(type) ? ObjectFlags.IsGenericObjectType : 0) |
-            (type.flags & (TypeFlags.InstantiableNonPrimitive | TypeFlags.Index | TypeFlags.TemplateLiteral | TypeFlags.StringMapping) && !isPatternLiteralType(type) ? ObjectFlags.IsGenericIndexType : 0);
+            (type.flags & (TypeFlags.InstantiableNonPrimitive | TypeFlags.Index | TypeFlags.StringMapping) && !isPatternLiteralType(type) ? ObjectFlags.IsGenericIndexType : 0);

```

Comment:
I feel like StringMapping / isPatternLiteralType could be dropped here, and instead just have `getGenericObjectFlags` return `getGenericObjectFlags((type as StringMappingType).type)` (as above), but I have been unable to find a bug that this fixes, even when combining the new intersection tests for the follow-up issues I found.